### PR TITLE
Pressure interpolation and a bunch of formatting changes

### PR DESCRIPTION
**Pull request by djgagne**
I have added pressure interpolation functionality in credit/interp.py and did a bunch of linting changes to clean up code using help from ruff.

### CODE REVIEW

File: credit/interp.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        surface_pressure (np.ndarray): Surface pressure in shape (y, x) and units Pa
+        temperature (np.ndarray): temperature in shape (levels, y, x) and units K
+        mixing_ratio (np.ndarray): mixing ratio in shape (levels, y, x) and units kg/kg.
+        model_a (np.ndarray): a coefficients at each model level being used in units of Pa.
+        model_b (np.ndarray): b coefficients at each model level being used (unitness).
+
+    Returns:
+        model_geoptential (np.ndarray): geopotential on model levels in shape (levels, y, x)
+    """
+    Rd = 287.0
```
Comment (yingkaisha) [inline credit/interp.py]: Use constants in `credit.physics_constants`

File: credit/interp.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    half_a = 0.5 * (model_a[:-1] + model_a[1:])
+    half_b = 0.5 * (model_b[:-1] + model_b[1:])
+    model_pressure = create_pressure_grid(surface_pressure, model_a, model_b)
+    half_pressure = create_pressure_grid(surface_pressure, half_a, half_b)
+    model_geopotential = np.zeros(model_pressure.shape, dtype=surface_pressure.dtype)
+    half_geopotential = np.zeros(half_pressure.shape, dtype=surface_pressure.dtype)
+    virtual_temperature = temperature * (1.0 + 0.608 * mixing_ratio)
```
Comment (yingkaisha) [inline credit/interp.py]: replace 0.608 with the following
```python
gamma = (RVGAS / RDGAS - 1.0)
virtual_temperature = temperature * (1 + gamma*mixing_ratio)
```

File: credit/postblock.py [hunk 3]
```diff
@@ -125,107 +124,146 @@ def forward(self, x):
...
+                    220,
+                    240,
+                    260,
+                    280,
+                    300,
+                    320,
+                    340,
+                ]
+            )
+
```
Comment (yingkaisha) [inline credit/postblock.py:181]: The original one
```python
x_demo = np.array([0, 20, 40, 60, 80, 100, 120, 140, 160, 180, 
                   200, 220, 240, 260, 280, 300, 320, 340])
```
seems easier to ready. Revert to the original formatting for this?

File: credit/postblock.py [hunk 4]
```diff
@@ -297,107 +340,141 @@ def forward(self, x):
...
```
Comment (yingkaisha) [inline credit/postblock.py:399]: Same as above

File: credit/output.py [hunk 5]
```diff
@@ -115,48 +117,54 @@ def save_netcdf_increment(darray_upper_air, darray_single_level, nc_filename, fo
...
         logger.info(f"Trying to save forecast hour {forecast_hour} to {nc_filename}")
 
         save_location = os.path.join(conf["predict"]["save_forecast"], nc_filename)
         os.makedirs(save_location, exist_ok=True)
-        
-        unique_filename = os.path.join(save_location, f"pred_{nc_filename}_{forecast_hour:03d}.nc")
+
+        unique_filename = os.path.join(
+            save_location, f"pred_{nc_filename}_{forecast_hour:03d}.nc"
+        )
```
Comment (djgagne) [inline credit/output.py:192]: Test

File: credit/output.py [hunk 6]
```diff
@@ -115,48 +117,54 @@ def save_netcdf_increment(darray_upper_air, darray_single_level, nc_filename, fo
...
+
         logger.info(f"Trying to save forecast hour {forecast_hour} to {nc_filename}")
 
         save_location = os.path.join(conf["predict"]["save_forecast"], nc_filename)
         os.makedirs(save_location, exist_ok=True)
-        
-        unique_filename = os.path.join(save_location, f"pred_{nc_filename}_{forecast_hour:03d}.nc")
+
+        unique_filename = os.path.join(
+            save_location, f"pred_{nc_filename}_{forecast_hour:03d}.nc"
```
Comment (djgagne) [inline credit/output.py:191]: Test

**Pull request merged**